tests/test_other.py

```diff
@@ -3784,14 +3799,14 @@ def test_bad_function_pointer_cast(self):
                 if relocate:
                   cmd += ['-s', 'RELOCATABLE=1'] # disables asm-optimized safe heap
                 print(cmd)
-                Popen(cmd).communicate()
+                run_process(cmd)
                 output = run_js('a.out.js', stderr=PIPE, full_output=True, assert_returncode=None)
                 if emulate_casts:
                   # success!
                   self.assertContained('Hello, world.', output)
                 else:
                   # otherwise, the error depends on the mode we are in
-                  if wasm and (relocate or emulate_fps):
+                  if self.is_wasm_backend() or (wasm and (relocate or emulate_fps)):
```

Comment:
Wasm backend always mismatches function signatures here?

Reply:
Yes, all modes have the mismatch but the specific error varies.  It seems that the wasm backend always produces this one.